pallets/settlement/src/lib.rs

```diff
@@ -1429,8 +1429,8 @@ impl<T: Config> Module<T> {
                         Error::<T>::MaxNumberOfNFTsPerLegExceeded
                     );
```

Comment:
I think we should include this check in the RPC endpoint, maybe add it to the `ensure_no_duplicate_nfts` function.